framework/console/Controller.php

```diff
@@ -680,7 +680,7 @@ public function getActionOptionsHelp($action)
 
     /**
      * @param Action $action
-     * @return \ReflectionMethod
+     * @return \ReflectionFunctionAbstract
```

Comment:
method return only ReflectionMethod instances

Reply:
We have a need to return either `ReflectionMethod` *or* `ReflectionFunction` in Craft CMS (see https://github.com/craftcms/cms/blob/1c66588bbf6b9745903df5f6553327a2549eaadb/src/console/Controller.php#L325-L329), and `getDocComment()` is defined in `ReflectionFunctionAbstract` so there’s no reason to force `ReflectionMethod` here.

Reply:
@brandonkelly it's not Craft CMS repo, getAction**Method**Reflection() return Reflection**Method**

Reply:
@brandonkelly because this method does not return ReflectionFunction as in Craft CMS and it's no need to extend PhpDoc in your CMS, that's extending not break LSP.